src/charm.py

```diff
@@ -662,6 +683,15 @@ def _on_peer_relation_changed(self, event: HookEvent) -> None:  # noqa: C901
 
         self.async_replication.handle_read_only_mode()
 
+    def _on_secret_changed(self, event: SecretChangedEvent) -> None:
+        """Handle the secret_changed event."""
+        if not self.unit.is_leader():
+            return
+
+        if admin_secret_id := self.config.system_users:  # noqa: SIM102
+            if admin_secret_id == event.secret.id:
+                self._update_admin_password(admin_secret_id)
```

Comment:
```suggestion
        if (admin_secret_id := self.config.system_users) and admin_secret_id == event.secret.id:
            self._update_admin_password(admin_secret_id)
```
The walrus op will be evaluated in the brackets.